code/go/0chain.net/chaincore/chain/state.go

```diff
@@ -146,6 +147,30 @@ func (c *Chain) UpdateState(ctx context.Context, b *block.Block, bState util.Mer
 	return c.updateState(ctx, b, bState, txn)
 }
 
+func (c *Chain) EstimateTransactionCost(ctx context.Context, b *block.Block, bState util.MerklePatriciaTrieI, txn *transaction.Transaction) int {
+	var (
+		clientState = CreateTxnMPT(bState) // begin transaction
+		sctx        = c.NewStateContext(b, clientState, txn, nil)
+	)
+
+	if txn.TransactionType == transaction.TxnTypeSmartContract {
+		var scData sci.SmartContractTransactionData
+		dataBytes := []byte(txn.TransactionData)
+		err := json.Unmarshal(dataBytes, &scData)
+		if err != nil {
+			logging.Logger.Error("Error while decoding the JSON from transaction",
+				zap.Any("input", txn.TransactionData), zap.Any("error", err))
+			return math.MaxInt32
```

Comment:
Return an error here should be more reasonable, so as when the TransactionType is incorrect.

Reply:
well, MaxInt is more like special value for a disabled transaction, the logic is simple, this transaction will be excluded from the block. We handled error already, while writing log, and now we have more common way to handle this situation. We already know how to handle this exception upper in the stack, without extending logic with   `if err != nil {.}` further, can reduce this boilerplate check, which can make sense

Reply:
fixed it